app/models/github_repo.rb

```diff
@@ -2,18 +2,27 @@ class GithubRepo < ApplicationRecord
   belongs_to :user
 
   serialize :info_hash, Hash
+
   validates :name, :url, :github_id_code, presence: true
-  validates :url, uniqueness: true
+  validates :url, url: true, uniqueness: true
   validates :github_id_code, uniqueness: true
 
+  scope :featured, -> { where(featured: true) }
+
   after_save :clear_caches
   before_destroy :clear_caches
 
-  def self.find_or_create(params)
-    repo = where(github_id_code: params[:github_id_code]).
+  # Update existing repository or create a new one with given params.
+  # Repository is searched by either GitHub ID or URL.
+  def self.upsert(user, **params)
```

Comment:
I called it `upsert` due to SQL lingo of "update or insert", let me know if `update_or_create` would be better